commands/src/main/java/org/jclouds/karaf/commands/compute/ComputeCommandWithOptions.java

```diff
@@ -85,7 +86,13 @@ protected ComputeService getComputeService() throws IOException {
       String providerValue = EnvHelper.getComputeProvider(provider);
       String apiValue = EnvHelper.getComputeApi(api);
       String identityValue = EnvHelper.getComputeIdentity(identity);
-      String credentialValue = EnvHelper.getComputeCredential(credential);
+      String credentialValue = null;
+      if (providerValue.equals("google-compute-engine")) {
+         credentialValue = EnvHelper.getGoogleCredentialFromJsonFile(credential);
+      } else {
+         credentialValue = EnvHelper.getComputeCredential(credential);
+      }
```

Comment:
You should expand the value of credential since the user may provide property or environment variable instead of a command line flag:

```java
String credentialValue = EnvHelper.getComputeCredential(credential);
if (providerValue.equals("google-compute-engine")) {
    credentialValue = EnvHelper.getGoogleCredentialFromJsonFile(credentialValue);
}
```